test/with_dummyserver/test_connectionpool.py

```diff
@@ -313,7 +327,7 @@ def test_nagle(self):
         self.addCleanup(conn.close)
         pool._make_request(conn, 'GET', '/')
         tcp_nodelay_setting = conn.sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY)
-        self.assertTrue(tcp_nodelay_setting)
+        assert tcp_nodelay_setting
```

Comment:
`assert tcp_nodelay_setting is True` would be better: we don't want to check if the value is [truthy](https://docs.python.org/3/library/stdtypes.html#truth-value-testing), we want to check that it is `True`, and it's not already obvious from the name that `tcp_nodelay_setting` is a boolean.

Reply:
`tcp_nodelay_setting` is not a boolean here, as shown by result of pytest 
```
tcp_nodelay_setting = conn.sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY)
>       assert tcp_nodelay_setting is True
E       AssertionError: assert 4 is True
```

Reply:
Sorry! Please forget about my comment, then!